Resources/public/Admin.js

```diff
@@ -332,22 +332,26 @@ var Admin = {
     },
 
     setup_collection_buttons: function(subject) {
+        var counter = jQuery(subject).closest('[data-prototype]').children().length;
```

Comment:
@EmmanuelVella: Just debuged the result of `counter` with my case on #2857. Got 0, should be 3.

Where is the `subject` pointing ? I suspect in addition that system to do a globally count (that not work) on all collection forms at the same time. We will have conflict with multiple collection form at the same time isn't it?